pkg/gds/backup.go

```diff
@@ -51,40 +50,61 @@ backups:
 		case <-ticker.C:
 		}
 
-		// Begin the backup process
-		start := time.Now()
-		log.Debug().Msg("starting backup")
-
-		// Conduct the backup, logging errors if needed
-		if err := s.db.(store.Backup).Backup(backupDir); err != nil {
-			// Do not continue if there was a backup error; all code in the rest of the
-			// loop should expect that the backup was successful.
-			// NOTE: using WithLevel and Fatal does not Exit the program like log.Fatal()
-			// this ensures that we issue a CRITICAL severity without stopping the server.
-			log.WithLevel(zerolog.FatalLevel).Err(err).Msg("could not backup database")
-			continue backups
-		} else {
-			log.Info().Dur("duration", time.Since(start)).Msg("backup complete")
+		// Execute the backup - error messages are logged in the Backup function so they
+		// are ignored here and are only returned from Backup for testing purposes.
+		s.Backup(backupDir)
+	}
+}
+
+// Backup performs a backup on behalf of the service. BackupManager calls this function
+// at a periodic interval to take a snapshot of the store to disk and to keep only the
+// configured number of archives.
+func (s *Service) Backup(path string) (err error) {
```

Comment:
This wasn't part of this PR - but the tests were breaking, and this was required to fix the tests.